src/storage/btree.c

```diff
@@ -1792,6 +1792,76 @@ static DISK_ISVALID btree_check_tree (THREAD_ENTRY * thread_p, const OID * class
 static DISK_ISVALID btree_check_by_btid (THREAD_ENTRY * thread_p, BTID * btid);
 static char *btree_unpack_mvccinfo (char *ptr, BTREE_MVCC_INFO * mvcc_info, short btree_mvcc_flags);
 
+static int btree_is_key_visible (THREAD_ENTRY * thread_p, BTID_INT * btid, PAGE_PTR pg_ptr,
+				 MVCC_SNAPSHOT * mvcc_snapshot, int slot_id, bool * is_visible, DB_VALUE * key_value);
+
+/*
+ *  btree_is_key_visible(): States if current key is visible or not.
+ *
+ *  thread_p(in): Thread entry.
+ *  btid(in): B-tree info.
+ *  pg_ptr(in):	Page pointer.
+ *  mvcc_snapshot(in): The MVCC snapshot.
+ *  slot_id(in) : Slot id to be looked for.
+ *  is_visible(out): True or False
+ *
+ *  return: error code if any error occurs.
+ */
+static int
+btree_is_key_visible (THREAD_ENTRY * thread_p, BTID_INT * btid, PAGE_PTR pg_ptr, MVCC_SNAPSHOT * mvcc_snapshot,
+		      int slot_id, bool * is_visible, DB_VALUE * key_value)
+{
+  RECDES record;
+  LEAF_REC leaf;
+  int num_visible = 0;
+  int key_offset = 0;
+  int ret = NO_ERROR;
+  int max_visible_oids = 1;
+  bool dummy_clear_key;
+
+  *is_visible = false;
+
+  if (mvcc_snapshot == NULL)
+    {
+      /* Early out. */
+      *is_visible = true;
+      return ret;
+    }
```

Comment:
I think it would be better to replace it with an asert statement.

assert(mvcc_snapshot != NULL);

Reply:
this routine is called with mvcc_snapshot (i. e. it's always true)

Reply:
That's why I think these codes are unnecessary.

Reply:
it looks so, I'm gonna fix it.